lib/components/OrderDialog/OrderDialog.tsx

```diff
@@ -95,64 +107,77 @@ export const OrderDialog: React.FC<OrderDialogProps> = ({
 
     setOrder(response.order)
     setOrderState(response.orderState)
-    setStage("order-progress")
+    setStage("progress")
+  }
+
+  const handleProgressContinue = async () => {
+    setLoading(true)
+    try {
+      // Simulate some async work
+      await new Promise((resolve) => setTimeout(resolve, 1000))
+
+      setSteps((currentSteps) => {
+        const activeStepIndex = currentSteps.findIndex((step) => step.active)
+        if (activeStepIndex === -1) return currentSteps
+
+        const newSteps = [...currentSteps]
+        // Complete current step
+        newSteps[activeStepIndex] = {
+          ...newSteps[activeStepIndex],
+          completed: true,
+          active: false,
+        }
+
+        // If this was the last step, move to checkout
+        if (activeStepIndex === currentSteps.length - 1) {
+          setTimeout(() => setStage("checkout"), 500) // Small delay to show completion
+        } else {
+          // Activate next step
+          newSteps[activeStepIndex + 1] = {
+            ...newSteps[activeStepIndex + 1],
+            active: true,
+          }
+        }
+
+        return newSteps
+      })
+    } finally {
+      setLoading(false)
+    }
```

Comment:
don't use `finally` (we should lint against it), most people don't understand how the flow control works